pkg/asset/manifests/ingress.go

```diff
@@ -108,13 +108,24 @@ func (ing *Ingress) generateClusterConfig(config *types.InstallConfig) ([]byte,
 			Name: "cluster",
 			// not namespaced
 		},
-		Spec: configv1.IngressSpec{
-			Domain: fmt.Sprintf("apps.%s", config.ClusterDomain()),
-		},
+		Spec: configv1.IngressSpec{},
 		Status: configv1.IngressStatus{
 			DefaultPlacement: defaultPlacement,
 		},
 	}
+	obj.Spec.Domain = fmt.Sprintf("apps.%s", config.ClusterDomain())
+	if config.AWS != nil && len(config.AWS.LBType) != 0 {
```

Comment:
Typically we will use a switch statement because other platforms may need to make changes as well. https://github.com/openshift/installer/blob/master/pkg/asset/manifests/infrastructure.go#L95 is a good example of how to do this.